### PR TITLE
EOS-25603: return skip for skipped tests

### PR DESCRIPTION
**Pull request by rkothiya**
As some of Motr STs are returning success without doing anything,
This patch attempts to fix that by configuring the tests
appropriately.

Signed-off-by: Rinku Kothiya <rinku.kothiya@seagate.com>

# Problem Statement
System Tests returning success without doing anything.

# Design
While reporting the result, we check if the test name is present in skip list then we return skip instead of success.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: utils/functions [hunk 1]
```diff
@@ -30,7 +87,17 @@ report_and_exit() {
     if [ $rc -eq 0 ]; then
         ## This message is used by Jenkins as a test success criteria;
         ## it must appear on stdout.
-        echo "$name: test status: SUCCESS"
+        if [[ "${LIBISAL_TESTS_SKIPLIST[*]}" =~ "$name"  ]] || [[ "${LIBFAB_TESTS_SKIPLIST[*]}" =~ "$name" ]];
+        then
```
Comment (rkothiya) [inline utils/functions:91]: if [[ "${LIBISAL_TESTS_SKIPLIST[*]}" =~ "$name" ]] || [[ $TRANSPORT = 'libfabric' 
 -a "${LIBFAB_TESTS_SKIPLIST[*]}" =~ "$name"]];

Comment (rkothiya) [reply utils/functions:87]: done

File: utils/functions [hunk 2]
```diff
@@ -1,5 +1,66 @@
 # -*- sh -*-
 
+#M0_SRC_DIR="$(readlink -f $0)"
+#M0_SRC_DIR="${M0_SRC_DIR%/*/*}"
+
+#. $M0_SRC_DIR/scripts/jenkins/build
+
+TRANSPORT=$(m0_default_xprt)
```
Comment (madhavemuri) [inline utils/functions:8]: Move this after definition of m0_default_xprt.

Comment (rkothiya) [reply utils/functions]: Done

**Comment by rkothiya** [comment]
As shown below the tests are skipped for libfabric job :
image

While it is not skipped for lnet job : 
image

**Pull request merged**